### PR TITLE
Check error instance for arguments test

### LINKED ISSUE #1087: Check the type of error instances in tests

### PR DESCRIPTION
**Pull request by IswaryaS**
Fixes #1087

First commit - I'd like verify that this is the expected change in the tests. Once that is done, will update in all the tests.

#### Checklist

- [X] I have read the documentation.
- [X] I have included a pull request description of my changes.
- [X] I have included some tests.
- [X] If it's a new feature, I have included documentation updates in both the README and the types.

### CODE REVIEW

**Comment by IswaryaS** [comment]
Hey, is this PR fine for the issue mentioned? Please read description as I've only changed for a single test right now..

**Comment by szmarczak** [comment]
Please don't duplicate PRs, you need to wait until the previous one gets merged.

**Comment by szmarczak** [comment]
Or, if the changes don't rely on the previous PR then you can remove those commits.

**Comment by IswaryaS** [comment]
Will remove those commits shortly.

File: test/arguments.ts
Comment (szmarczak) [inline test/arguments.ts:105]: Can you do this as well please?

File: test/arguments.ts
Comment (szmarczak) [inline test/arguments.ts:311]: Same here :)

File: test/arguments.ts
Comment (szmarczak) [inline test/arguments.ts:453]: Here as well

**Review by szmarczak — APPROVED** [review]
LGTM, it would be awesome if you could the TODOs mentioned above as well :heart:

**Comment by szmarczak** [comment]
Thanks a lot! 🥇

**Comment by IswaryaS** [comment]
@szmarczak Hey. I'm looking to pick up some more tasks, maybe slightly more challenging. If there's any change in particular you'd like to be done for this repository, let me know and I'll take a look. That's if you don't mind and willing to help me out if needed.

**Comment by szmarczak** [comment]
Sure, thanks for your interest! Here you go: https://github.com/sindresorhus/got/issues/1462

**Pull request merged**
